### DISCUSSION TITLE
[Vue warn]: Component provided template option but runtime compilation is not supported in this build of Vue. Configure your bundler to alias "vue" to "vue/dist/vue.esm-bundler.js".

### DISCUSSION THREAD

**Discussion by likymm** [nuxt/nuxt#20381] [Questions] [answered]
Hello guys,

Hope you are all doing well. Just want to know if someone already tried to create their own interpolation using [dynamic component](https://vuejs.org/guide/essentials/component-basics.html#dynamic-components). Already been fixing this for days but still can't figure it out.
```vue
<template>
  <h4>Hi!</h4>
  <component
    :is="{
      template: dynamicTemp,
    }"
  />
</template>
<script setup lang="ts">
const firstName = "Joe";
const useInputContent = "<h3>Welcome {name}</h3>";

// Replace {name} to "Joe"
const dynamicTemp = useInputContent.replace("{name}", firstName);
</script>
```

```ts
// nuxt.config.ts
export default defineNuxtConfig({
  ssr: false,
});
```
```json
{
  "private": true,
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "devDependencies": {
    "nuxt": "^3.4.1"
  }
}
```

When I'm running that, I'm getting this warning: [Vue warn]: Component provided template option but runtime compilation is not supported in this build of Vue. Configure your bundler to alias "vue" to "vue/dist/vue.esm-bundler.js".

Please suggest if there are other ways or a workaround. I'm just new on Vue and Nuxt.

Don't want to create a bug first because I'm not sure if this issue is coming from Nuxt 3.

Will really appreciate the help. Thanks

**Comment by huang-julien** [comment, marked as answer]
Hi you need to enable the `experimental.runtimeVueCompiler`. But in your case, it might be better to use a `v-html` attr

**Comment by likymm** [comment]
Yey! Enabling it worked for me. #13843. I actually need this instead of `v-html` as I also pass components in my HTML string.

Here's my working nuxt config for reference.
```ts
// nuxt.config.ts
export default defineNuxtConfig({
  ssr: false,
  hooks: {
    'vite:extendConfig': (config, { isClient }) => {
      if (isClient) {
        (config!.resolve!.alias! as any).vue = 'vue/dist/vue.esm-bundler';
      }
    },
  },
});
```

Thanks @huang-julien you're the man!

**Reply by Xowap** [reply]
Thanks for sharing this.

When I do that, my `nuxt build` finally works but on the other hand the SSR (which I can't disable) hangs forever and does not serve the pages I ask.

So I'm guessing this breaks runtime compilation of templates SSR in production mode...

At this point I'm starting to think that I'm screwed.

**Reply by huang-julien** [reply]
strange, can you open an issue with a reproduction please ? I'll take a look
